Automattic/simplenote-ios
# Updated AppIcon assets to work with new ios 18 styles

### Fix
In iOS 18 we are given the ability to customize the look of app icons on our Home Screen.  The app icons should be able to respond to light mode, dark mode, and custom schemes.  For some reason the `AppIcon` and the `AppIcon-Beta` icons were not compatible with this new feature, but the `AppIcon-Sustainer` was.

So in this PR I have updated the `AppIcon` and `AppIcon-Beta` to work with the new Home Screen customizations.

fixes #1658
### Test
1. Launch Simplenote from Xcode. Must be running the iOS 18 Beta, so you will need an Xcode beta to test this
2. After the app launches go to the Home Screen, long press on the background and then tap Edit > customize. 
3. Try all of the following settings and make sure the app icon updates to the correct theme:
- Small - Light
- Small - Dark
- Small -Automatic
- Small - Tinted
- Large - Light
- Large - Dark
- Large -Automatic
- Large - Tinted 

4. In Xcode go to the project navigator and select the Simplenote target > General.  Scroll down to App Icon and Home Screen. 
5. Next to App Icon tap on the three dots and find Debug.  Change the value from `AppIcon-Beta` to `AppIcon`
6. Repeat steps 1-3 for the release app icon

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in d3adb3ef with:
> 
> > Added markdown support

If the changes should not be included in release notes, add a statement to this section. For example:

> These changes do not require release notes.
